integration/tests/cook/test_basic.py

```diff
@@ -892,45 +920,48 @@ def jobs_query():
 
             util.wait_until(jobs_query, util.all_instances_done)
             # retry all jobs in the group
-            util.retry_jobs(self.cook_url, retries=2, groups=[group_uuid])
+            util.retry_jobs(self.cook_url, retries=2, groups=[group_uuid], failed_only=failed_only)
             # wait for some job to start
             util.wait_until(group_query, util.group_some_job_started)
-            # ensure none of the jobs are still in a failed state
-            job_data = util.query_jobs(self.cook_url, job=jobs)
-            self.assertEqual(200, job_data.status_code)
-            for job in job_data.json():
-                job_details = f'Job details: {json.dumps(job, sort_keys=True)}'
-                self.assertNotEqual('failed', job['state'], job_details)
+            # return final job details to caller for assertion checks
+            return util.query_jobs(self.cook_url, assert_response=True, job=jobs).json()
         finally:
             # ensure that we don't leave a bunch of jobs running/waiting
             util.kill_groups(self.cook_url, [group_uuid])
 
+    def test_group_change_killed_retries(self):
+        jobs = self._help_test_group_change_killed_retries(failed_only=True)
+        # ensure none of the jobs are still in a failed state
+        for job in jobs:
+            self.assertNotEqual('failed', job['state'], f'Job details: {json.dumps(job, sort_keys=True)}')
+
+    def test_group_change_killed_retries_failed_only(self):
```

Comment:
How come the test that sets `failed_only` to `True` is called `test_group_change_killed_retries` and the test that sets it to `False` is called `test_group_change_killed_retries_failed_only`?

Reply:
That was backwards! I fixed it. Thanks!